esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -3121,11 +3126,33 @@ void CFileSprayEx::searchDropZoneFiles(IEspContext& context, const char* server,
         if (!fname.length())
             continue;
 
+        if (di->isDir())
+        {
+            StringBuffer fullPath(dir);
+            addPathSepChar(fullPath).append(fname);
+            if (getDropZoneScopePermissions(context, dropZoneName, server, fullPath) < SecAccess_Read)
+                continue;
+        }
+
+        StringBuffer sPath;
+        const char* ptr = fname.str();
+        const char* pName = strrchr(ptr, pathSep);
+        if (pName)
+        {
+            sPath.append(pName - ptr, ptr);
+            fname.set(pName + 1);
+
+            StringBuffer fullPath(dir);
+            addPathSepChar(fullPath).append(sPath);
+            if (getDropZoneScopePermissions(context, dropZoneName, server, fullPath) < SecAccess_Read)
```

Comment:
I suspect I'm misreading.. but line 3133 has already checked scope perms if it's a directory,

this (lines 3137 to 3149) seems to be checking if di->getName is a file or directory within a subdir (by seeing if it has a path sep. in it), then building up the path up to the parent of that file, and checking that dir's perms.

But if it's a subdir (not a file within a subdir) - then wouldn't 3133 have already checked it?

Reply:
The directoryFiles() returns the files under the dir. The line 3133 only checks scope perms if the di is a folder. If the di is not a folder (ex. /dir/subfolder1/file1), the code here checks the scope access for the subfolders (ex. /dir/subfolder1).

Reply:
I should add the code to do this only if di is not a folder.